src/operation-poller.ts

```diff
@@ -0,0 +1,74 @@
+import { Queue } from "./throttler/queue";
+import * as api from "./api";
+import * as FirebaseError from "./error";
+
+export interface OperationPollerOptions {
+  pollerName?: string;
+  apiOrigin: string;
+  apiVersion: "v1" | "v1beta1";
+  operationResourceName: string;
+  backoff?: number;
+  masterTimeout?: number;
+}
+
+const DEFAULT_INITIAL_BACKOFF_DELAY_MILLIS = 250;
+const DEFAULT_MASTER_TIMEOUT_MILLIS = 30000;
+
+export class OperationPoller {
```

Comment:
One other suggestion: how about making the class (and helper method) generic such that it takes the type of the expected result. It can default to `any` if not specified, but that lets us be more type specific when we want to be.

Reply:
Great idea, +1

Reply:
Made the class and the exported function generic